labeler-module/pkg/labeler/labeler.go

```diff
@@ -229,13 +235,41 @@ func (l *Labeler) getDriverLabelForNode(nodeName string) (string, error) {
 		}
 
 		if podutil.IsPodReady(pod) {
-			return "true", nil
+			return LabelValueTrue, nil
 		}
 	}
 
 	return "", nil
 }
 
+// getKataLabelForNode detects if Kata is enabled on the specified node
+// Returns "true" if Kata is enabled, "false" if not, or error if detection fails
+func (l *Labeler) getKataLabelForNode(nodeName string) (string, error) {
```

Comment:
I'm not sure I follow the changes. Since we have 2 daemonsets, can't we pass an additional arg to avoid the API call and cache here?

Reply:
OK, so 
Labeler detects Kata → Sets node labels (API calls happen here - already cached)
DS use node labels for scheduling → Pods get deployed with --kata-enabled flag
Syslog pods receive the flag → No detection needed, just informational